### PR TITLE
Small improvements to sphinx docs

### PR DESCRIPTION
**Pull request by kanderso-nrel**
## Description

This PR:
- addresses a handful of small issues in the sphinx docs (missing parameter descriptions, incorrect parameter names, etc)
- enables intersphinx for a few packages
- updates the sphinx version used to build the docs

## Checklist

The following items must be addressed before the code can be merged. 
Please don't hesitate to ask for help if you are unsure of how to accomplish any of the items. 
*You are free to remove any checklist items that do not apply or add additional items that are 
not on this list*

- ~Closes #xxx~
- ~Added new API functions to `docs/api.rst`~
- ~Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings~
- [ ] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- ~Non-API functions clearly documented with docstrings or comments as necessary~
- ~Added tests to cover all new or modified code~
- [x] Pull request is nearly complete and ready for detailed review

### CODE REVIEW

File: pvanalytics/features/shading.py [hunk 1]
```diff
@@ -321,6 +321,9 @@ def fixed(ghi, daytime, clearsky, interval=None, min_gradient=2):
     interval : int, optional
         Interval between data points in minutes. If not specified the
         interval is inferred from the frequency of the index of `ghi`.
+    min_gradient : int, default 2
+        Threshold value for the morphological gradient.
+        TODO: what is a threshold value for a morphological gradient?
```
Comment (kanderso-nrel) [inline pvanalytics/features/shading.py:326]: Someone please suggest a description for this parameter, ideally mentioning why one might want to change its value from the default.  Also is "int" actually the appropriate type?

File: pvanalytics/quality/gaps.py [hunk 2]
```diff
@@ -335,7 +335,7 @@ def complete(series, minimum_completeness=0.333, freq=None):
 
     See Also
     --------
-    :py:func:`completeness_score`
+    completeness_score
```
Comment (kanderso-nrel) [inline pvanalytics/quality/gaps.py:338]: No need for the `:py:func:` business in "See Also"; sphinx links them automatically anyway.

File: pvanalytics/quality/time.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ def shifts_ruptures(event_times, reference_times, period_min=2,
     time of day. To establish the expected midday time, calculate solar
     transit time in time of day. This function detects shifts in the
     difference between the observed and expected midday times, and
-    returns (here I'm unclear what is being returned)
+    returns (TODO: here I'm unclear what is being returned)
```
Comment (kanderso-nrel) [inline pvanalytics/quality/time.py:143]: Maybe just remove this sentence entirely, since the return value is described elsewhere?

**Review by kanderso-nrel — COMMENTED** [review]
A few questions and comments below.  

Unrelated: thoughts on switching to "squash and merge" as the default for merging PRs?  It's what we do in pvlib and rdtools -- the idea is that individual commits in a PR are not interesting on their own, so might as well squash them on merge so that master has a nice commit history.

**Comment by kanderso-nrel** [comment]
Test failures seem unrelated; I'll take a look locally and might open a separate PR to resolve if I figure it out.

**Comment by cwhanse** [comment]
> Test failures seem unrelated; I'll take a look locally and might open a separate PR to resolve if I figure it out.

Looks like pvanalytics tests are compatible with pvlib<0.9.0, where `orientation_strategy` was a parameter for `pvlib.modelchain.ModelChain`. Since 0.9.0 that parameter has been removed.

From pvlib what's new for v0.9.0:

The orientation_strategy parameter has been removed from the various pvlib.modelchain.ModelChain constructors and surface_tilt, surface_azimuth are now required parameters for pvlib.modelchain.basic_chain() (GH1028, GH1181)

File: pvanalytics/features/shading.py (hunk 1 above)
Comment (cwhanse) [reply pvanalytics/features/shading.py:326]: I don't know that we can define that with any more clarity. Type should be float.

https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.ndimage.morphology.morphological_gradient.html

**Comment by cwhanse** [comment]
I support switching to "squash and merge". @wfvining ?

**Comment by wfvining** [comment]
Squash and merge is fine with me. Previously I was doing squash and rebase, which also results in a nice history. I could see a good argument in favor of including a merge commit though, so it is fine with me.

File: pvanalytics/quality/time.py (hunk 3 above)
Comment (wfvining) [reply pvanalytics/quality/time.py:143]: Axe it. I'm not sure what it is doing there in the first place.

**Comment by cwhanse** [comment]
Thanks @kanderso-nrel

**Pull request merged**